lib/identity/validation/validatePublicKeysFactory.js

```diff
@@ -0,0 +1,73 @@
+const { PublicKey } = require('@dashevo/dashcore-lib');
+
+const IdentityPublicKey = require('../IdentityPublicKey');
+
+const ValidationResult = require('../../validation/ValidationResult');
+
+const InvalidIdentityPublicKeyTypeError = require(
+  '../../errors/InvalidIdentityPublicKeyTypeError',
+);
+const InvalidIdentityPublicKeyDataError = require(
+  '../../errors/InvalidIdentityPublicKeyDataError',
+);
+
+/**
+ * Validate public keys (factory)
+ *
+ * @param {validateDuplicatePublicKeys} validateDuplicatePublicKeys
+ *
+ * @return {validatePublicKeys}
+ */
+function validatePublicKeysFactory(validateDuplicatePublicKeys) {
+  /**
+   * Validate public keys type and data and duplication
+   *
+   * @typedef validatePublicKeys
+   *
+   * @param {IdentityPublicKey[]|RawIdentityPublicKey[]} publicKeys
+   *
+   * @return {ValidationResult}
+   */
+  function validatePublicKeys(publicKeys) {
+    const result = new ValidationResult();
+
+    const rawPublicKeys = publicKeys
+      .map((publicKey) => {
+        if (publicKey instanceof IdentityPublicKey) {
+          return publicKey.toJSON();
+        }
+
+        return publicKey;
+      });
+
+    // validate duplication
+    result.merge(
+      validateDuplicatePublicKeys(rawPublicKeys),
```

Comment:
I meant to put everything in the same function.

Reply:
@shumkov I would be a pretty fat function, isn't it better to keep them apart?

Reply:
Discussed online and decided to move all validation logic (including schema) to this function.